lib/trino-orc/src/main/java/io/trino/orc/stream/LongInputStreamV2.java

```diff
@@ -408,6 +409,29 @@ public void next(short[] values, int items)
         }
     }
 
+    @Override
+    public void next(byte[] values, int items) throws IOException
+    {
+        int offset = 0;
+        while (items > 0) {
+            if (used == numLiterals) {
+                numLiterals = 0;
+                used = 0;
+                readValues();
+            }
+
+            int chunkSize = min(numLiterals - used, items);
+            for (int i = 0; i < chunkSize; i++) {
+                long literal = literals[used + i];
+                byte value = SignedBytes.saturatedCast(literal);
```

Comment:
Should be
```
byte value = (byte) literal;
if (literal != value) {
    throw new OrcCorruptionException(input.getOrcDataSourceId(), "Decoded value out of range for a 8bit number");
}
```

Reply:
Same for other usages of `saturatedCast`

Reply:
Thanks for review, i will squash my commits and open other PR.